### PR TITLE
refactor: remove ClusterAccess wrapper class

### LINKED ISSUE #2164: Remove ClusterAccess class
### Component

JKube Kit

### Description

After #2163 / #2155 the `ClusterAccess` class is just a wrapper around the `ClusterConfiguration` class. We should remove `ClusterAccess` altogether.

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix #2164

Related to https://github.com/eclipse-jkube/jkube/pull/3142

+ Move ClusterAccess and ClusterConfiguration classes from `jkube-kit/config/resource` -> `jkube-kit/common` so that they're accessible to JKubeConfiguration
+ Add new field `clusterConfiguration` in JKubeConfiguration for passing in Cluster configuration
+ Add new field `currentContext` to ClusterConfiguration in order to pass current context
+ Remove ClusterAccess from JKubeServiceHub constructor
   - ClusterAccess is not required in the constructor as ClusterConfiguration
    is being passed in JKubeConfiguration
   - Modify tests relying on ClusterAccess mocks to use KubernetesMockServer instead
+ Remove ClusterAccess field from Mojos/Tasks; initialize a local instance where necessary
+ Update tests relying on ClusterAccess mocked instance to use KubernetesMockServer instead.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3200 (_2024-06-26T11:21:02Z_)

:gear: [JKube E2E Tests (9678697279)](https://github.com/eclipse-jkube/ci/actions/runs/9678697279) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3200)](https://github.com/eclipse-jkube/ci/actions/runs/9678697279/job/26703031369)

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeConfiguration.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public class JKubeConfiguration implements Serializable {
   private static final long serialVersionUID = 7459084747241070651L;
 
   private JavaProject project;
+  private transient ClusterConfiguration clusterConfiguration;
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeConfiguration.java:42]: Why transient?

**Comment by manusa** [comment]
I don't mind if it's in this PR or in a follow up PR, but ClusterConfiguration->ClusterAccess should be removed from JKubeServiceHub constructor since now it's part of JKubeConfiguration

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeConfiguration.java (hunk 1 above)
Comment (rohanKanojia) [reply jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeConfiguration.java:42]: IDE was showing a warning as ClusterConfiguration is not serializable . I think we do not want to expose this field via Serialization

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java [hunk 2]
```diff
@@ -176,7 +174,7 @@ protected GeneratorContext.GeneratorContextBuilder initGeneratorContextBuilder()
         .prePackagePhase(false)
         .useProjectClasspath(kubernetesExtension.getUseProjectClassPathOrDefault())
         .sourceDirectory(kubernetesExtension.getBuildSourceDirectoryOrDefault())
-        .openshiftNamespace(StringUtils.isNotBlank(kubernetesExtension.getNamespaceOrNull()) ? kubernetesExtension.getNamespaceOrNull() : clusterAccess.getNamespace())
+        .openshiftNamespace(StringUtils.isNotBlank(kubernetesExtension.getNamespaceOrNull()) ? kubernetesExtension.getNamespaceOrNull() : new ClusterAccess(initClusterConfiguration()).getNamespace())
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:177]: Why do we need ClusterAccess here?
Have you checked what ClusterAccess.getNamespace does?

**Review by manusa — CHANGES_REQUESTED** [review]
I think we're not on the same page here.

It's basically easier to try and tackle https://github.com/eclipse-jkube/jkube/issues/2164 **first** and then do whatever is needed here.

If you carefully analyze what you're doing at the moment is just adding more wrappers around clusterconfiguration, so basically you're making things worse.

If you simply remove `ClusterAccess` from the picture you'll see that everything falls naturally into place.

Do you prefer for me to bootstrap these changes?

**Review by manusa — CHANGES_REQUESTED** [review]
https://github.com/rohanKanojia/jkube/pull/123

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**